File: backend/controllers/notificationController.js
const notificationModel = require('../models/notification');

async function getList(req, res) {
  try {
    const { id: userId } = req.user;
    const filters = {
      isRead: req.query.isRead !== undefined ? req.query.isRead === 'true' : undefined,
      type: req.query.type || undefined,
      page: parseInt(req.query.page) || 1,
      pageSize: parseInt(req.query.pageSize) || 20
    };
    
    const result = await notificationModel.getByUserId(userId, filters);
    res.json({ success: true, data: result.data, total: result.total, page: result.page, pageSize: result.pageSize });
  } catch (error) {
    console.error('getList error:', error);
    res.status(500).json({ success: false, message: String(error.message || error) });
  }
}

async function getUnreadCount(req, res) {
  try {
    const { id: userId } = req.user;
    const count = await notificationModel.getUnreadCount(userId);
    res.json({ success: true, data: { count } });
  } catch (error) {
    console.error('getUnreadCount error:', error);
    res.status(500).json({ success: false, message: String(error.message || error) });
  }
}

async function markAsRead(req, res) {
  try {
    await notificationModel.markAsRead(req.params.id);
    res.json({ success: true, message: '已标记为已读' });
  } catch (error) {
    console.error('markAsRead error:', error);
    res.status(500).json({ success: false, message: String(error.message || error) });
  }
}

async function markAllAsRead(req, res) {
  try {
    const { id: userId } = req.user;
    await notificationModel.markAllAsRead(userId);
    res.json({ success: true, message: '全部已标记为已读' });
  } catch (error) {
    console.error('markAllAsRead error:', error);
    res.status(500).json({ success: false, message: String(error.message || error) });
  }
}

async function remove(req, res) {
  try {
    const { id: userId } = req.user;
    const success = await notificationModel.remove(req.params.id, userId);
    if (!success) {
      return res.status(404).json({ success: false, message: '通知不存在' });
    }
    res.json({ success: true, message: '通知删除成功' });
  } catch (error) {
    console.error('remove error:', error);
    res.status(500).json({ success: false, message: String(error.message || error) });
  }
}

module.exports = {
  getList,
  getUnreadCount,
  markAsRead,
  markAllAsRead,
  remove
};

File: jkstore_new/README/oneflow-api/src/main/java/com/oneflow/api/notification/NotificationController.java
package com.oneflow.api.notification;

import com.oneflow.api.auth.CurrentUser;
import com.oneflow.api.common.ApiResponse;
import com.oneflow.api.security.AuthSupport;
import java.util.LinkedHashMap;
import java.util.Map;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestHeader;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/notifications")
public class NotificationController {

    private final NotificationRepository notificationRepository;
    private final AuthSupport authSupport;

    public NotificationController(NotificationRepository notificationRepository, AuthSupport authSupport) {
        this.notificationRepository = notificationRepository;
        this.authSupport = authSupport;
    }

    @GetMapping
    public ResponseEntity<?> list(
            @RequestHeader(value = "Authorization", required = false) String authorization,
            @RequestParam(value = "isRead", required = false) Boolean isRead,
            @RequestParam(value = "type", required = false) String type,
            @RequestParam(value = "page", defaultValue = "1") int page,
            @RequestParam(value = "pageSize", defaultValue = "20") int pageSize) {
        CurrentUser user = requireUser(authorization);
        ResponseEntity<ApiResponse<String>> denied = requireNotificationView(user);
        if (denied != null) return denied;
        Map<String, Object> result = notificationRepository.findByUserId(user.getId(), isRead, type, page, pageSize);
        Map<String, Object> response = new LinkedHashMap<String, Object>();
        response.put("success", true);
        response.putAll(result);
        return ResponseEntity.ok(response);
    }

    @GetMapping("/unread-count")
    public ResponseEntity<?> unreadCount(@RequestHeader(value = "Authorization", required = false) String authorization) {
        CurrentUser user = requireUser(authorization);
        ResponseEntity<ApiResponse<String>> denied = requireNotificationView(user);
        if (denied != null) return denied;
        Map<String, Object> data = new LinkedHashMap<String, Object>();
        data.put("count", notificationRepository.unreadCount(user.getId()));
        return ResponseEntity.ok(ApiResponse.success(data));
    }

    @PutMapping("/{id}/read")
    public ResponseEntity<?> markAsRead(
            @RequestHeader(value = "Authorization", required = false) String authorization,
            @PathVariable String id) {
        CurrentUser user = requireUser(authorization);
        ResponseEntity<ApiResponse<String>> denied = requireNotificationView(user);
        if (denied != null) return denied;
        notificationRepository.markAsRead(id, user.getId());
        ApiResponse<String> response = ApiResponse.success(null);
        response.setMessage("已标记为已读");
        return ResponseEntity.ok(response);
    }

    @PutMapping("/read-all")
    public ResponseEntity<?> markAllAsRead(@RequestHeader(value = "Authorization", required = false) String authorization) {
        CurrentUser user = requireUser(authorization);
        ResponseEntity<ApiResponse<String>> denied = requireNotificationView(user);
        if (denied != null) return denied;
        notificationRepository.markAllAsRead(user.getId());
        ApiResponse<String> response = ApiResponse.success(null);
        response.setMessage("全部已标记为已读");
        return ResponseEntity.ok(response);
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<?> remove(
            @RequestHeader(value = "Authorization", required = false) String authorization,
            @PathVariable String id) {
        CurrentUser user = requireUser(authorization);
        ResponseEntity<ApiResponse<String>> denied = requireNotificationView(user);
        if (denied != null) return denied;
        if (!notificationRepository.delete(id, user.getId())) {
            return ResponseEntity.status(HttpStatus.NOT_FOUND)
                    .body(ApiResponse.<String>failure(404, "通知不存在"));
        }
        ApiResponse<String> response = ApiResponse.success(null);
        response.setMessage("通知删除成功");
        return ResponseEntity.ok(response);
    }

    private CurrentUser requireUser(String authorization) {
        return authSupport.parseBearerUser(authorization);
    }

    private ResponseEntity<ApiResponse<String>> requireNotificationView(CurrentUser user) {
        if (user == null) {
            return ResponseEntity.status(HttpStatus.UNAUTHORIZED)
                    .body(ApiResponse.<String>failure(401, "未登录或登录已过期"));
        }
        if (!authSupport.hasPermission(user, "notification:view")) {
            return ResponseEntity.status(HttpStatus.FORBIDDEN)
                    .body(ApiResponse.<String>failure(403, "Permission denied: notification:view is required"));
        }
        return null;
    }
}
